订阅失败,返回错误如下.

4-26 17:55:27 [ERROR] logging | Exception in ASGI application

Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\bot.py", line 17, in <module>
    nonebot.run()
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\nonebot\__init__.py", line 273, in run
    get_driver().run(*args, **kwargs)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\nonebot\drivers\fastapi.py", line 187, in run
    uvicorn.run(
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\uvicorn\main.py", line 569, in run
    server.run()
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\uvicorn\server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 640, in run_until_complete
    self.run_forever()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 607, in run_forever
    self._run_once()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 1919, in _run_once
    handle._run()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
> File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 419, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\fastapi\applications.py", line 271, in __call__
    await super().__call__(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\applications.py", line 125, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 79, in __call__
    raise exc
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 21, in __call__
    raise e
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\routing.py", line 706, in __call__
    await route.handle(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\routing.py", line 443, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\fastapi\applications.py", line 271, in __call__
    await super().__call__(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\applications.py", line 125, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 79, in __call__
    raise exc
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 21, in __call__
    raise e
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\routing.py", line 706, in __call__
    await route.handle(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\fastapi\routing.py", line 237, in app
    raw_response = await run_endpoint_function(
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\fastapi\routing.py", line 163, in run_endpoint_function
    return await dependant.call(**values)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\nonebot_bison\admin_page\api.py", line 153, in get_target_name
    return {"targetName": await check_sub_target(platformName, T_Target(target))}
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\nonebot_bison\apis.py", line 12, in check_sub_target
    return await platform_manager[platform_name].get_target_name(client, target)
  File "C:\Users\Administrator\Desktop\NBBOOT\FistBot\.venv\Lib\site-packages\nonebot_bison\platform\bilibili.py", line 79, in get_target_name
    res_data = json.loads(res.text)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\json\decoder.py", line 340, in decode
    raise JSONDecodeError("Extra data", s, end)

添加BILIBILI订阅IP输入ID的时候出现,提示服务器错误.

--- a/nonebot_bison/platform/bilibili.py
+++ b/nonebot_bison/platform/bilibili.py
@@ -250,7 +250,7 @@ class Bilibililive(StatusChange):
         cls, client: AsyncClient, target: Target
     ) -> Optional[str]:
         res = await client.get(
-            "https://api.bilibili.com/x/space/acc/info", params={"mid": target}
+            "https://api.bilibili.com/x/space/wbi/acc/info", params={"mid": target}
         )
         res_data = json.loads(res.text)
         if res_data["code"]:

--- a/nonebot_bison/platform/bilibili.py
+++ b/nonebot_bison/platform/bilibili.py
@@ -74,7 +74,7 @@ class Bilibili(NewMessage):
         cls, client: AsyncClient, target: Target
     ) -> Optional[str]:
         res = await client.get(
-            "https://api.bilibili.com/x/space/acc/info", params={"mid": target}
+            "https://api.bilibili.com/x/space/wbi/acc/info", params={"mid": target}
         )
         res.raise_for_status()
         res_data = res.json()

--- a/tests/test_config_manager_add.py
+++ b/tests/test_config_manager_add.py
@@ -423,7 +423,7 @@ async def test_add_with_bilibili_target_parser(app: App, init_scheduler):
     from nonebot_bison.platform.bilibili import Bilibili
 
     ak_list_router = respx.get(
-        "https://api.bilibili.com/x/space/acc/info?mid=161775300"
+        "https://api.bilibili.com/x/space/wbi/acc/info?mid=161775300"
     )
     ak_list_router.mock(
         return_value=Response(200, json=get_json("bilibili_arknights_profile.json"))

